packages/react-priority-overflow/src/useOverflowMenu.ts

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { useId, useIsomorphicLayoutEffect } from '@fluentui/react-utilities';
+import { useOverflowContext } from './overflowContext';
+import { useOverflowCount } from './useOverflowCount';
+import { DATA_OVERFLOW_MENU } from './constants';
+
+export function useOverflowMenu<TElement extends HTMLElement>(id?: string) {
+  const elementId = useId('overflow-menu', id);
+  const overflowCount = useOverflowCount();
+  const registerItem = useOverflowContext(v => v.registerItem);
+  const updateOverflow = useOverflowContext(v => v.updateOverflow);
+  const ref = React.useRef<TElement>(null);
+  const isOverflowing = overflowCount > 0;
+
+  useIsomorphicLayoutEffect(() => {
+    let deregisterItem: () => void = () => null;
+    const element = ref.current;
+    if (element) {
+      deregisterItem = registerItem({
+        element,
+        id: elementId,
+        priority: Infinity,
+      });
+      element.setAttribute(DATA_OVERFLOW_MENU, '');
+    }
+
+    return () => {
+      element?.removeAttribute(DATA_OVERFLOW_MENU);
+      deregisterItem();
+    };
+  }, [registerItem, isOverflowing, elementId]);
+
+  React.useEffect(() => {
```

Comment:
Is it okay to use `useEffect` there? Should it be `useLayoutEffect`?

Reply:
you're right, since it's dealing with the DOM node and requires measurement of the bounding box, it's safer to do it in a layout effect before browser paint